Fixing the current placement system

clusterfack opened the pull request:
The current placement system is buggy in every possible way
'
1. Every placement scheme except one requires placement within a certain range (wtf)
2. There is no circle to show where you can place objects within range, though there is meant to be
3. The collision system prevents 3/4ths of the placement modes from placing anything
4. Everyone of the systems duplicates a fuckload of code for no reason
5. Once you spawn an entity, they must either use the same render or some shit because the ones you spawned will flicker along with the ghost image of the one you are about to spawn
6. Most of the placement mode names make absolutely no sense whatsoever

This addresses most of those issues, and brings the working placement modes from the 1 that it currently is up to about 6 or 7 out of 8.

RemieRichards commented on SS14.Client/Placement/PlacementMode.cs:
```diff
@@ -66,5 +67,33 @@ public Sprite GetDirectionalSprite(string baseSprite)
             return GetSprite((baseSprite + "_" + pManager.Direction.ToString()).ToLowerInvariant());
         }
+
+        public virtual bool RangeRequired()
```
this should probably just go IN ```RangeCheck()```, so that it always succeeds if it's not required;
Shortens down all the ifs from ```if(RangeRequired() && !RangeCheck())``` to just ```if(!RangeCheck())```

PJB3005 commented on SS14.Client/Placement/Modes/AlignNearby.cs:
```diff
@@ -0,0 +1,47 @@
+        {
+        }
+
+        public override bool RangeRequired()
```
Make this a property instead of a method:
```cs
public override bool RangeRequired => true;
```

clusterfack commented:
Update:
- 3/4ths of the placement modes are fixed, the others I'm not sure what their precise functions are meant to be anyway so we can come back to those later, they aren't as crucial.
- Fixed a bug where the sprite for the sprite to be placed was being drawn from the global list of sprites, having its color modified, and then being used. Leading to the clientside list of sprites having a permanently changed sprite for every version of that sprite. *we had ought to look into this topic at some point*
Fixes some other various problems with collision, the suggestions people made, changed some names to more accurately reflect what they do.
